salt/vault/service.sls

```diff
@@ -0,0 +1,33 @@
+{% from "vault/map.jinja" import vault with context %}
+
+vault_service_init_file_managed:
+  file.managed:
+    - name: {{ vault.service.path }}
+    - source: {{ vault.service.source }}
+    - template: jinja
+    - defaults:
+        dev_configs: {{ vault.dev_configs }}
+
+{%- if not vault.dev_mode %}
+
+manage_selinux_mode:
+  selinux.mode:
+    - name: permissive
+
```

Comment:
this block can be simplified using the opposite form of `require`... `require_in`:

```
{%- if not vault.dev_mode %}
manage_selinux_mode:
  selinux.mode:
    - name: permissive
    - require_in:
      - service: vault_service_running
{%- endif %+

vault_service_running:
  service.running:
    - name: vault
    - enable: True
    - reload: True
```